packages/frontend/src/components/UserSettings/SlackSettingsPanel/index.tsx

```diff
@@ -51,35 +57,48 @@ const SlackSettingsPanel: FC = () => {
             enabled: isValidSlack,
         });
     const { mutate: deleteSlack } = useDeleteSlack();
-    const { mutate: updateNotificationChannel } =
-        useUpdateSlackNotificationChannelMutation();
+    const { mutate: updateCustomSettings } =
+        useUpdateSlackAppCustomSettingsMutation();
 
-    const form = useForm<{ notificationChannel: string | null }>({
+    const form = useForm<SlackAppCustomSettings>({
         initialValues: {
             notificationChannel: null,
+            appProfilePhotoUrl: null,
         },
     });
 
-    const { setFieldValue } = form;
+    const { setFieldValue, onSubmit } = form;
 
     useEffect(() => {
         if (data?.notificationChannel) {
             setFieldValue('notificationChannel', data.notificationChannel);
         }
-    }, [data?.notificationChannel, setFieldValue]);
+
+        if (data?.appProfilePhotoUrl) {
+            setFieldValue('appProfilePhotoUrl', data.appProfilePhotoUrl);
+        }
+    }, [data?.appProfilePhotoUrl, data?.notificationChannel, setFieldValue]);
```

Comment:
I've been trying to get the right pattern to manage initial/async state with mantine forms a while ago. here it is:

file: `packages/frontend/src/components/UserSettings/ProfilePanel/index.tsx`

![CleanShot 2024-06-10 at 16 43 00@2x](https://github.com/lightdash/lightdash/assets/962095/6b2330d3-be83-4660-a91a-b94531ed2748)

Reply:
this makes the form reset properly too